### DISCUSSION TITLE
LauncherInerceptor -- can I leverage this to run tests I've bundled with an webapp?

### DISCUSSION THREAD

**Discussion by JohnZ1385** [junit-team/junit-framework#3473] [Q&A] [answered]
just out of curiosity has anyone managed to successfully leverage the LauncherInterceptor to execute tests residing within a WAR file? 
As an exploded WAR I am able to find all of the tests with the default classloader but if I deploy the application as an archive (i.e. .. a WAR deployment to the server) it appears the LauncherDiscoveryRequest never finds any of the tests.

See my stackoverflow post for full detail:
https://stackoverflow.com/questions/77084650/invoking-junit-within-a-webapp

**Comment by marcphilipp** [comment]
Sounds closely related to https://github.com/junit-team/junit5/issues/1724

**Comment by JohnZ1385** [comment, marked as answer]
@marcphilipp thanks for pointing me in the right direction
I followed this comment: https://github.com/junit-team/junit5/issues/1724#issuecomment-1402001181
for the time being and used a runtime annotation to find the appropriate classes and everything works fine now

**Comment by JohnZ1385** [comment]
see https://github.com/junit-team/junit5/issues/1724
